app/src/lib/notifications/toasts.ts

```diff
@@ -22,7 +22,8 @@ export function showToast(toast: Toast) {
 }
 
 export function showError(title: string, err: any) {
```

Comment:
Would it be possible to strongly type this `err`? Or is that prevented by `err` somehow being `string | Error`?

From all I can tell, the backend [should always provide a map with `code` and `message`](https://github.com/gitbutlerapp/gitbutler/blob/5a9962bc5a602ab932a36445715ccae03a2016c3/crates/gitbutler-tauri/src/error.rs#L60-L68), but apparently at least in this case, that doesn't happen.

Reply:
<img width="398" alt="Screenshot 2024-04-24 at 17 04 40" src="https://github.com/gitbutlerapp/gitbutler/assets/63622/b456dc27-6632-4763-93b1-1b8501edc7e6">

```
2024-04-24T15:04:26.171529Z ERROR add_project: crates/gitbutler-tauri/src/projects.rs:27: error=Error(Bare repositories are unsupported

Caused by:
    bare repositories are not supported) path="/Users/byron/dev-gb/bare"
```

This looks like the error type is not a string even when it arrives in the frontend.

Do errors come in from other places than [the IPC code here](https://github.com/gitbutlerapp/gitbutler/blob/dac89e41a980d3e5acfef50f624d8539d5f0079a/app/src/lib/backend/ipc.ts#L26-L31)?